src/pages/builders/challenges.tsx

```diff
@@ -1,22 +1,81 @@
+import { useState } from 'react'
+import type { GetServerSideProps, NextPage } from 'next'
+import classNames from 'classnames'
 import { config } from '@/lib/config'
+import Phases from '@/data/phase'
 import { Head } from '@/components/head/Head'
 import { Header } from '@/components/layout/header/Header'
 import { Footer } from '@/components/layout/footer/Footer'
-import type { GetServerSideProps, NextPage } from 'next'
 import type { Config } from '@/types/config.type'
+import type { ChallengeDTO } from '@/data/phase/dto.type'
 
-export type ChallengesProps = Pick<Config, 'title' | 'keywords' | 'description' | 'urls'>
+export type ChallengesProps = Pick<Config, 'title' | 'keywords' | 'description' | 'urls' | 'phases'>
+
+type CategoryTabProps = {
+  toggleTab: () => void
+  selectedTab: ChallengeDTO
+  challenge: ChallengeDTO
+}
+
+const CategoryTab: React.FC<CategoryTabProps> = ({ toggleTab, challenge, selectedTab }) => (
+  <li
+    className={classNames('okp4-nemeton-web-page-challenges-categories-tab', {
+      active: challenge === selectedTab
+    })}
+    onClick={toggleTab}
+  >
+    {challenge.challengeCategorie}
+  </li>
+)
 
 const Challenges: NextPage<ChallengesProps> = props => {
-  const { urls } = props
+  const { urls, phases } = props
+  const [categorieChallenges] = useState<ChallengeDTO[]>(
+    Phases(phases)[Phases(phases).length - 1].challenges ?? []
+  )
```

Comment:
WHat's the purpose of this state?
If there is no setter, please use `useMemo` instead